File: backend/src/main/java/com/sai/backend/domain/ai/dto/request/AiFeedbackRequestDto.java
package com.sai.backend.domain.ai.dto.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.springframework.web.multipart.MultipartFile;

@Getter
@Setter
@AllArgsConstructor
public class AiFeedbackRequestDto {
    private String code;
    private MultipartFile logImage;
    private MultipartFile resultImage;
    private String memo;
    private String threshold;
}

File: app/SAI/SAI/SAI.Application/Dto/AiFeedbackRequestDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SAI.SAI.Application.Dto
{
    public class AiFeedbackRequestDto
    {
        [Required] public string code { get; set; } = string.Empty;   // 코드 원문
        [Required] public string logImage { get; set; } = string.Empty;   // 학습/실행 로그 이미지 (상대 경로)
        [Required] public string resultImage { get; set; } = string.Empty;   // 결과 이미지 (상대 경로)
        [Required] public string memo { get; set; } = string.Empty;

        [Required] 
        [Range(0.01, 1.00)]
        public double threshold { get; set; }
    }
}
